stratum/hal/lib/tdi/tdi_table_manager.cc

```diff
@@ -1198,197 +1198,195 @@ ::util::Status TdiTableManager::ReadMeterEntry(
+          ->set_policer_spec_cbs(static_cast<int64>(cfg[i].cburst));
+      result.mutable_config()
+          ->mutable_policer_meter_config()
+          ->set_policer_spec_eir(static_cast<int64>(cfg[i].pir));
+      result.mutable_config()
+          ->mutable_policer_meter_config()
+          ->set_policer_spec_ebs(static_cast<int64>(cfg[i].pburst));
+      result.mutable_counter_data()->mutable_green()->set_byte_count(
+          static_cast<int64>(cfg[i].greenBytes));
+      result.mutable_counter_data()->mutable_green()->set_packet_count(
+          static_cast<int64>(cfg[i].greenPackets));
+      result.mutable_counter_data()->mutable_yellow()->set_byte_count(
+          static_cast<int64>(cfg[i].yellowBytes));
+      result.mutable_counter_data()->mutable_yellow()->set_packet_count(
+          static_cast<int64>(cfg[i].yellowPackets));
+      result.mutable_counter_data()->mutable_red()->set_byte_count(
+          static_cast<int64>(cfg[i].redBytes));
+      result.mutable_counter_data()->mutable_red()->set_packet_count(
+          static_cast<int64>(cfg[i].redPackets));
+
+      *resp.add_entities()->mutable_meter_entry() = result;
     }
 
-    return ::util::OkStatus();
+    VLOG(1) << "ReadMeterEntry resp " << resp.DebugString();
+    if (!writer->Write(resp)) {
+      return MAKE_ERROR(ERR_INTERNAL) << "Write to stream for failed.";
+    }
   }
 
-  ::util::Status SetPktModMeterConfig(TdiPktModMeterConfig config,
-                                    const ::p4::v1::MeterEntry& meter_entry) {
-    config.meter_prof_id =
-        meter_entry.config().policer_meter_config().policer_meter_prof_id();
-    config.cir_unit =
-        meter_entry.config().policer_meter_config().policer_spec_cir_unit();
-    config.cburst_unit =
-        meter_entry.config().policer_meter_config().policer_spec_cbs_unit();
-    config.pir_unit =
-        meter_entry.config().policer_meter_config().policer_spec_eir_unit();
-    config.pburst_unit =
-        meter_entry.config().policer_meter_config().policer_spec_ebs_unit();
-    config.cir = meter_entry.config().policer_meter_config().policer_spec_cir();
-    config.cburst =
-        meter_entry.config().policer_meter_config().policer_spec_cbs();
-    config.pir = meter_entry.config().policer_meter_config().policer_spec_eir();
-    config.pburst =
-        meter_entry.config().policer_meter_config().policer_spec_ebs();
-    config.greenBytes = meter_entry.counter_data().green().byte_count();
-    config.greenPackets = meter_entry.counter_data().green().packet_count();
-    config.yellowBytes = meter_entry.counter_data().yellow().byte_count();
-    config.yellowPackets = meter_entry.counter_data().yellow().packet_count();
-    config.redBytes = meter_entry.counter_data().red().byte_count();
-    config.redPackets = meter_entry.counter_data().red().packet_count();
+  return ::util::OkStatus();
+}
 
-    return ::util::OkStatus();
-  }
+::util::Status SetPktModMeterConfig(TdiPktModMeterConfig config,
```

Comment:
```suggestion
::util::Status SetPktModMeterConfig(TdiPktModMeterConfig& config,
```
Otherwise, the method will initialize a _copy_ of `config`, and your changes will be lost when it returns.

**BUG**